src/libraries/System.Numerics.Vectors/tests/GenericVectorTests.cs

```diff
@@ -846,6 +846,72 @@ public void VectorSingleEqualsNaNTest()
             var nan = new Vector<float>(float.NaN);
             Assert.True(nan.Equals(nan));
         }
+
+        [Fact]
+        public void VectorDoubleEqualsNonCanonicalNaNTest()
+        {
+            var cast = BitConverter.DoubleToUInt64Bits(double.NaN);
+            var one = BitConverter.DoubleToUInt64Bits(1);
+            var sign = one ^ BitConverter.DoubleToUInt64Bits(-1);
+            var mant = one ^ BitConverter.DoubleToUInt64Bits(1.2);
+            var nan = new Vector<double>(double.NaN);
+            var pnan = new Vector<double>(BitConverter.UInt64BitsToDouble(cast ^ sign));
```

Comment:
if you wanted a definite `+qnan`, then `double.CopySign(double.NaN, +0.0)` or using the raw bits directly is the "safest"
likewise for `-qnan`, then `double.CopySign(double.NaN, -0.0)`

Reply:
love it! much cleaner.